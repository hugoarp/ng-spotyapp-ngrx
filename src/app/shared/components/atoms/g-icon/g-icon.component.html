<i class="material-icons g-icon"> {{ icon }} </i>
