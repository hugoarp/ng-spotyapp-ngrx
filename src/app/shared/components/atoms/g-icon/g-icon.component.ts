import { Component, Input, OnInit } from '@angular/core';

@Component({
  selector: 'app-g-icon',
  templateUrl: './g-icon.component.html',
  styleUrls: ['./g-icon.component.scss'],
})
export class GIconComponent implements OnInit {
  @Input() icon: string = '';

  constructor() {}

  ngOnInit(): void {}
}
