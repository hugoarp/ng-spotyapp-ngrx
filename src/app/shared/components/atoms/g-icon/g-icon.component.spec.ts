import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { GIconComponent } from './g-icon.component';

describe('GIconComponent', () => {
  let component: GIconComponent;
  let fixture: ComponentFixture<GIconComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [GIconComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    }).compileComponents();

    fixture = TestBed.createComponent(GIconComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  it('debe recibir un valor de tipo string en el input icon', () => {
    const icon = 'home';
    component.icon = icon;
    fixture.detectChanges();
  });
});
