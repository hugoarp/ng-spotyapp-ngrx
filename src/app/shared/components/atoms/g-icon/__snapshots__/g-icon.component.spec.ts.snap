// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIconComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <i
    class="material-icons g-icon"
  >
      
  </i>
</div>
`;
